init.lua

```diff
@@ -32,119 +32,88 @@ aliases[" "] = "space"
 
 
 
-
-local function getXOffset(s)
-	local xpos = settings.x_position
-
-	if type(xpos) == "number" then
-		return xpos + s.geometry.x
-	elseif xpos == "left" then
-		return s.geometry.x + settings.x_offset
-	elseif xpos == "right" then
-		return s.geometry.x + s.geometry.width - modewibox[s].width - settings.x_offset
-	elseif xpos == "center" then
-		return s.geometry.x + s.geometry.width / 2 - modewibox[s].width / 2 + settings.x_offset
-	end
-	return 0
-end
-
-
-local function getYOffset(s)
-	local ypos = settings.y_position
-
-	if type(ypos) == "number" then
-		return ypos + s.geometry.y
-	elseif ypos == "top" then
-		return s.geometry.y + settings.y_offset
-	elseif ypos == "bottom" then
-		return s.geometry.y + s.geometry.height - modewibox[s].height - settings.y_offset
-	elseif ypos == "center" then
-		return s.geometry.y + s.geometry.height / 2 - modewibox[s].height / 2 + settings.y_offset
-	end
-	return 0
-end
-
-local function calculate_position(s)
-	local minwidth, minheight = modewidget[s]:fit({dpi=96}, s.geometry.width,
-		s.geometry.height)
-	modewibox[s].width = minwidth + 1;
-	modewibox[s].height = math.max(settings.height, minheight)
-
-	modewibox[s].x = getXOffset(s)
-	modewibox[s].y = getYOffset(s)
-end
-
-local function update_settings()
-	for s, value in pairs(modewibox) do
-		calculate_position(s)
-		value.opacity = settings.opacity
-	end
-end
-
-
 function modalbind.init()
-	awful.screen.connect_for_each_screen(function(s)
-		modewidget[s] = wibox.widget.textbox()
-		modewidget[s]:set_align("left")
-		if beautiful.fontface then
-			modewidget[s]:set_font(beautiful.fontface .. " " .. (beautiful.fontsize + 4))
-		end
+	local modewibox = wibox({
+			ontop=true,
+			visible=false,
+			x=0,
+			y=0,
+			width=1,
+			height=1,
+			opacity=defaults.opacity,
+			bg=beautiful.modebox_bg or "#000",
+			shape=gears.shape.round_rect,
+			type="toolbar"
+	})
+
+	modewibox:setup({
```

Comment:
I agree that the declarative wiboxes look cleaner (the old code was also ported from awesome 3.2). And it does not need to run once per screen :+1: